src/xtgeo/surface/_regsurf_cube_window_attributes.py

```diff
@@ -0,0 +1,422 @@
+            "xori": self.cube.xori,
+            "yori": self.cube.yori,
+            "zori": resampled_depth_arr.min(),
+            "rotation": self._reduced_cube.rotation,
+            "yflip": self._reduced_cube.yflip,
+        }
+
+        self._refined_cube = Cube(
+            **common_params,
+            values=resampled_arr.astype(np.float32),
+        )
+        self._refined_depth_cube = Cube(
+            **common_params,
+            values=resampled_depth_arr.astype(np.float32),
+        )
+
+    def _depth_mask(self):
+        """Set nan values outside the interval defined by the upper + lower surface.
+
+        In addition, set nan values where the thickness is less than the minimum.
+
+        """
+        print("Depth mask...")
+
+        depth_2d_upper_exp = np.expand_dims(self._upper.values, 2)
+        depth_2d_lower_exp = np.expand_dims(self._lower.values, 2)
+        mask_2d_exp = np.expand_dims(self._min_thickness_mask.values, 2)
+
+        depth_3d = self._refined_depth_cube.values
+
+        self._refined_cube.values = np.where(
+            (depth_3d < depth_2d_upper_exp)
+            | (depth_3d > depth_2d_lower_exp)
+            | (mask_2d_exp == 0),
+            np.nan,
+            self._refined_cube.values,
+        ).astype(np.float32)
+
+        # similar for reduced cubes with original resolution
+        depth_3d = self._reduced_depth_cube.values
+
+        self._reduced_cube.values = np.where(
+            (depth_3d < depth_2d_upper_exp)
+            | (depth_3d > depth_2d_lower_exp)
+            | (mask_2d_exp == 0),
+            np.nan,
+            self._reduced_cube.values,
+        ).astype(np.float32)
+
+    def _compute_statistical_attribute_surfaces(self) -> None:
+        """Compute the attributes and update the dict that holds all such maps"""
+
+        print("Compute statistical attributes...")
+
+        def _add_to_attribute_map(attr_name, values) -> None:
+            """Compute the attribute map and add to result dictionary."""
+            attr_map = self._upper.copy()
+            attr_map.values = np.ma.masked_invalid(values)
+
+            # now resample to the original input map
+            attr_map_resampled = self.this_surface.copy()
+            attr_map_resampled.resample(attr_map)
+
+            attr_map_resampled.values = np.ma.masked_where(
+                self.this_surface.values.mask, attr_map_resampled.values
+            )
+
+            self._result_attr_maps[attr_name] = attr_map_resampled
+
+        def _safe_nanvar(arr_slice):
+            """Variance, incl. handling when number of non-NaN elements is <= 1."""
+            if np.sum(~np.isnan(arr_slice)) > 1:
+                return np.nanvar(arr_slice)
+            return np.nan
+
+        cref = self._refined_cube.values
+        with np.errstate(invalid="ignore"):
+            for attr in STAT_ATTRS:
+                if attr == "mean":
+                    values = np.nanmean(cref, axis=2)
```

Comment:
Maybe `STAT_ATTRS` should be a mapping?

```python
STAT_ATTRS = {
    "mean": lambda cref: np.nanmean(cref, axis=2),
    ...
}
# ...
for attr, fn in STAT_ATTRS.items():
    if attr in ("list", "to", "skip"):
        continue

    _add_to_attribute_map(attr, fn(cref))
```

I see not all of them could be lambda functions, but it might still be more succint